pex/pex.py

```diff
@@ -617,10 +617,47 @@ def execute_script(self, script_name):
         dist_script = get_script_from_distributions(script_name, dists)
         if not dist_script:
             return "Could not find script {!r} in pex!".format(script_name)
-        TRACER.log("Found script {!r} in {!r}.".format(script_name, dist))
-        return self.execute_content(
-            dist_script.path, dist_script.read_contents(), argv0=script_name
-        )
+        TRACER.log("Found script {!r} in {!r}.".format(script_name, dist_script.dist))
+
+        contents = dist_script.read_contents()
+        shebang = contents.split("\n", 1)[0]
+        try:
+            interpreter = PythonInterpreter.from_shebang(shebang)
+            if interpreter and interpreter.version_string != self.interpreter.version_string:
+                TRACER.log(
```

Comment:
Neither this check and log nor the except clause log are useful afaict. When scripts are installed in venvs its expected that their shebangs are re-written to point to the current venv interpreter: https://setuptools.readthedocs.io/en/latest/deprecated/distutils/setupscript.html?highlight=scripts#installing-scripts. In the case of a PEX file, this is the executing with the PEX file itself which is what `execute_content` does. So I think a few changes are in order:
1. Python script detection should work exactly like it does for distutils / setuptools - this will address the shebang parsing non-robustness observation aobve.
2. Python script detection only need present a bool result.

Its probably the best to add this parsing / bool directly to DistributionScript which already encapsulates finding these things:
https://github.com/pantsbuild/pex/blob/37c9ff0fa815ac1f03a73d3f2e33fb20352c8843/pex/finders.py#L18-L43

It looks like [poetry](https://python-poetry.org/docs/pyproject/#scripts) and [flit](https://flit.readthedocs.io/en/latest/pyproject_toml.html#scripts-section) only support console scripts so the setuptools / distutils behavior fully defines scripts afaict.